source/grid/grid_out.cc

```diff
@@ -2139,7 +2139,13 @@ GridOut::write_svg(const Triangulation<2, 2> &tria, std::ostream &out) const
           double h;
 
           if (n != 1)
-            h = .6 - (index / (n - 1.)) * .6;
+            {
+              // The assert is a workaround a compiler bug in ROCm 5.7 which
+              // evaluated index/(n-1) when n == 1 in debug mode. When adding
+              // the assert the ratio is not evaluated.
+              Assert((n - 1.) != 0., ExcInvalidState());
```

Comment:
Any reason for a float comparison?
```suggestion
              Assert((n - 1) != 0, ExcInvalidState());
```

Reply:
Yes, that's the compiler bug. The compiler evaluates  `(n - 1.)` to zero and then throws a floating point exception because it divides by zero.